pkg/util/objectwatcher/objectwatcher.go

```diff
@@ -69,32 +69,31 @@ func (o *objectWatcherImpl) Create(clusterName string, desireObj *unstructured.U
 	// users should resolve the conflict in person.
 	clusterObj, err := dynamicClusterClient.DynamicClientSet.Resource(gvr).Namespace(desireObj.GetNamespace()).Create(context.TODO(), desireObj, metav1.CreateOptions{})
 	if err != nil {
+		existObj, err := dynamicClusterClient.DynamicClientSet.Resource(gvr).Namespace(desireObj.GetNamespace()).Get(context.TODO(), desireObj.GetName(), metav1.GetOptions{})
+		if err != nil {
+			return fmt.Errorf("failed to get exist resource(kind=%s, %s/%s) in cluster %v: %v", desireObj.GetKind(), desireObj.GetNamespace(), desireObj.GetName(), clusterName, err)
+		}
 		// The 'IsAlreadyExists' conflict may happen in following known scenarios:
 		// - 1. In a reconcile process, the execution controller successfully applied resource to member cluster but failed to update the work conditions(Applied=True),
 		//   when reconcile again, the controller will try to apply(by create) the resource again.
 		// - 2. The resource already exist in the member cluster but it's not created by karmada.
 		if apierrors.IsAlreadyExists(err) {
```

Comment:
> if promote a nodeport service from member cluster to control plane,apierrors.IsAlreadyExists(err) is true

Ah?    line 73  `err != nil` has returned ,  line 76 `err` it must be `nil` ,therefore `apierrors.IsAlreadyExists(err)` must be false. 

right ?

Reply:
> > if promote a nodeport service from member cluster to control plane,apierrors.IsAlreadyExists(err) is true
> 
> Ah? line 73 `err != nil` has returned , line 76 `err` it must be `nil` ,therefore `apierrors.IsAlreadyExists(err)` must be false.
> 
> right ?

sorry,I misunderstand before.Excution_controller will create a resource in member cluster instead of update it by the manifest in works .So if a reosurce is exist in member cluster,`apierrors.IsAlreadyExists(err)` will return true.However, if a reource is a nodeport service,create if will return notInvalid err.

Reply:
yes ,l also found this problems , if a reource is a nodeport service,create it will return [provided port is already allocated](https://github.com/kubernetes/kubernetes/blob/master/pkg/registry/core/service/portallocator/allocator.go#L43)